ARCHITECTURE.md

```diff
@@ -0,0 +1,194 @@
+# Standard Design and Architecture
+
+At the time of writing, almost a year of exploratory and freestyle project history has passed.
+Fortunately, it is not necessary for further understanding, so I'll spare you that.
+This document, though, lays out the design, architecture and direction of Standard.
+
+If the topics discussed herein are dear to you, please take it as an invitation to get involved.
+
+This design document shall be stable and amendments go through a proper process of consideration.
+
+## Overview
+
+Standard is a collection of functionality and best practices (_"framework"_) to bootstrap and sustain the automatable sections of the Software Delivery Lifecycle (SDLC) _efficiently_ with the power of Nix and Flakes.
+In particular, Standard is a _Horizontal\* Integration Framework_ which integrates _vertical\*_ tooling.
+
+> <sub>We occasionally adapt concepts from non-technical contexts. This is one instance.</sub>
+>
+> _Vertical Tooling_ does one thing and does it well in a narrow scope (i.e "vertical").
+>
+> _Horizontal Tooling_ stitches vertical tooling together to a polished whole.
+
+What is being integrated are the end-to-end automatable sections of the SDLC.
+For these we curate a collection of functionality, tools and best practices.
+
+An SDLCs _efficiency_ on the is characterized by two things.
+
+Firstly, by adequate _lead time_ which is the amount of time it takes to set up an initial version of the software delivery pipeline.
+It needs to be _adequate_ rather than _just fast_, as it takes place in the context of a team.
+Rather than for speed, they need optimization for success.
+For example, a process needs to be documented & explained and your team needs to be trained on it.
+Standard encourages incremental adoption in order to leave enough space for these paramount activities.
+If you're in a hurry and your team is onboard, though, you still can jumpstart the implementation.
+
+Secondly, efficient SDLCs are characterized by short _cycle times_ which is the amount of time it takes for a designed feature to be shipped to production.
+Along this journey, we encounter our scope (more on it below):
+
+- aspects of the _development_ environment;
+- the packaging pipeline that produces artifacts;
+- and continuous processes integrating the application lifecycle.
+
+Hence, the goal of Standard is to:
+
+- Enable easy and incremental adoption
+- Optimize the critical path that reduces your SDLC's cycle time.
+
+Additionally, unlike similar projects, we harness the power of Nix & Flakes to ensure reproducibility.
```

Comment:
maybe expand on this a bit? It doesn't seem like a lot until you actually use it. How does the reproducible nature of Nix enhance the SDLC in a unique way? It is hard for us who are already so familiar with it to articulate that clearly to someone who is not (since it seems so obvious), but doing so could make all the difference.

Reply:
I think an more than shallow understanding of "reproducibility" would have to be assumed if we don't want this to be about it, instead. But we want this to be about _efficient_ SDLC, earmarked by desirable setup & cycle times.

Nix & Flakes is a pervasive theme of the architecture, but its also rather sparsely mentioned for the prominence it takes in the implementation. That's because its an implementation detail to those goals. An important one, because it affords _reproducibility_ which improves _cycle time_ in a variety of ways.

We still underline _reproducibility_ for its (rather collateral) software supply chain security connotation. Strictly speaking, we could achieve that though with something else than Nix.

And I believe part of this project (Standard) is an attempt to get Nix on-par with peer solutions, which Nix isn't currently. It's a niche and a satellite at the same time. Which is why I ultimately believe we have a lot of head-space to dovetail well with emerging standards in this field of software supply chain security.

This dovetailing would be covered by the scope, but we don't point it out more than the mere reference to _reproducibility_, because we haven't even started a serious discussion yet on how that all would have to look like.

Part of the truth is also that Nix is not the _only_ approach to reproducibility, so we could make this easily a footgun, too.

For these reasons, I prefer not to lash out on Nix/Reproducibility here, even though we've been thought that this is why we're using Nix. But is that actually true after it being used as the initial carrot to help us jump on the learning curve?